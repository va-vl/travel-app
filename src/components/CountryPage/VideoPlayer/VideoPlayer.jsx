import * as React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/styles';

const VideoPlayer = ({ videoUrl }) => {
  const classes = styles();

  return (
    <div className={`block ${classes.root}`}>
      <iframe
        width="853"
        height="480"
        src={`https://www.youtube-nocookie.com/embed/${videoUrl}`}
        frameBorder="0"
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
        title="Embedded youtube"
      />
    </div>
  );
};

VideoPlayer.propTypes = {
  videoUrl: PropTypes.string.isRequired,
};

export default VideoPlayer;
