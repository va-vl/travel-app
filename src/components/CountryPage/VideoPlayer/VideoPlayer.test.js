import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import VideoPlayer from './VideoPlayer';

test('component VideoPlayer renders', () => {
  const div = document.createElement('div');
  ReactDOM.render(<VideoPlayer videoUrl="https://testUrl.com"></VideoPlayer>, div);
});

test('render correctly VideoPlayer component', () => {
  const VideoPlayerComponent = TestRenderer.create(<VideoPlayer videoUrl="https://testUrl.com" />).toJSON();
  expect(VideoPlayerComponent).toMatchSnapshot();
});
