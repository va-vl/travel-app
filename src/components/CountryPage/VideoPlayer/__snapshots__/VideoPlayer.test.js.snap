// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly VideoPlayer  1`] = `
<div
  className="block makeStyles-root-1"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="480"
    src="https://www.youtube-nocookie.com/embed/undefined"
    title="Embedded youtube"
    width="853"
  />
</div>
`;

exports[`render correctly VideoPlayer component 1`] = `
<div
  className="block makeStyles-root-1"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="480"
    src="https://www.youtube-nocookie.com/embed/https://testUrl.com"
    title="Embedded youtube"
    width="853"
  />
</div>
`;

exports[`render correctly text component 1`] = `
<div
  className="block makeStyles-root-1"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="480"
    src="https://www.youtube-nocookie.com/embed/undefined"
    title="Embedded youtube"
    width="853"
  />
</div>
`;
